### DISCUSSION TITLE
[Feature Request]: Apply theming to GTK4/GNOME 45 apps

### DISCUSSION THREAD

**Discussion by JohnRTitor** [JaKooLit/Hyprland-Dots#223] [Q&A] [answered]
## Feature Request

Currently themes like Tokyo Night and Cappachino are not being applied to GNOME 45 apps like Nautilus. And Chrome (with GTK on) do not respect the theme like Firefox does. 

### Proposed Solution

For GTK4 apps, include a theme so they can get the blur and transparent look.

**Comment by JohnRTitor** [comment, marked as answer]
Note: For GTK4 apps, one needs to force a particular theme using `GTK_THEME` env var, or use a patched version of libadwaita. See [GTK on Arch Wiki](https://wiki.archlinux.org/title/GTK#GTK_3_and_GTK4).

I tested and this breaks dynamic switching between dark/light mode and other themes, since changing system environment variables mid session is not possible. It needs at least a relogin.

Unless @JaKooLit can figure out other means..

**Comment by JaKooLit** [comment]
> Note: For GTK4 apps, one needs to force a particular theme using `GTK_THEME` env var, or use a patched version of libadwaita. See [GTK on Arch Wiki](https://wiki.archlinux.org/title/GTK#GTK_3_and_GTK4).
> 
> I tested and this breaks dynamic switching between dark/light mode and other themes, since changing system environment variables mid session is not possible. It needs at least a relogin.
> 
> Unless @JaKooLit can figure out other means..

gtk4 apps has been quite challenging to apply the 3rd party themes tbh...

**Comment by JohnRTitor** [comment]
An alternative solution would be to use a "neutral theme" which would look good on both light and dark mode. But that won't be possible :(

On another note, for QT apps, I still see some uneven "non" rounded corner UI elements, can you check that perhaps

**Comment by dhelgerson** [comment]
white-sur gtk has this solved. you could look into how they did it

**Comment by JohnRTitor** [comment]
You mean [this](https://github.com/vinceliuice/WhiteSur-gtk-theme?tab=readme-ov-file#fix-for-libadwaita-not-perfect)? Like I said earlier, it's possible to theme GTK 4 apps, but it will break switching to other themes and switching dark/light mode without a logout/restart.
